Mautic form fixes

7ochem opened the pull request:
I am issuing this pull request to fix several issues:

1. I am sending a big amount of data. What cURL is doing, is setting the `Expect: 100-continue` header (see "[When curl sends 100-continue](https://gms.tf/when-curl-sends-100-continue.html)"). The result of that is that the response is containing two header sections. See below. My proposal would this response
    ```
          'header' => string 'HTTP/1.1 100 Continue' (length=21)
          'content' => string 'HTTP/1.1 302 Found
    Date: Wed, 17 Apr 2019 11:41:44 GMT
    Content-Type: text/html; charset=UTF-8
    Transfer-Encoding: chunked
    Connection: keep-alive
    Set-Cookie: 89f00040d7b0ca501905216bb6d1f780=e11ba72237d8a42fc224c91404a29a42fc237073; path=/; secure; HttpOnly
    Set-Cookie: mautic_session_id=deleted; expires=Thu, 01-Jan-1970 00:00:01 GMT; Max-Age=0; path=/; secure
    Set-Cookie: mautic_device_id=z8abb7jmtwkzvvply0q25v9; expires=Thu, 16-Apr-2020 11:41:44 GMT; Max-Age=31536000; path=/; secure
    [...]
    ```
    Possibly fixing #7? (going out on a limb here :grin:)

2. `curl_exec()` can return a non-string value (such as `false`). My changes would populate the response `'header'` and `'content'` array elements with `null` and if a string response has been returned by `curl_exec()` then it will populate those elements with strings (like the result would be before).
    This prevents the line `list($header, $content) = explode("\r\n\r\n", $result, 2);` from giving a _PHP Notice_ "Undefined offset: 1" as `$result` does not contain `"\r\n\r\n"` and offset 1 cannot be found by `list()` to populate `$content`
    Actually solving #6 

3. The `prepareRequest()` method initialises the `$request` array with the value `['header']` which results in an array element with integer key `0` and value `'header'`. I believe the intent was to initialise the array with an empty headers array: `['header' => []]` as later on this is done: `$request['header'][] = ...` without initialising the `$request['header']` element as an array (is allowed in PHP, but could emit a _PHP Notice_)

_P.S. I see tests fail on files that haven't been touched by my  PR. Can this be fixed or not taken into account with this PR?_

escopecz commented:
Thanks! It make sense. Could you please check if you could fix why is Travis complaining? And it would make me more comfortable to merge this if you'd added tests covering this scenario to https://github.com/escopecz/mautic-form-submit/blob/master/tests/Mautic/FormTest.php

7ochem commented:
What a fast response! How great

> [...] Could you please check if you could fix why is Travis complaining?

This is due to PHPCS errors in the file `src/HttpHeader.php` which I did not change (so has already been corrupt on master) and PHPStan giving an error on the lines below the ones I have changed (so probably also already on master?).

> And it would make me more comfortable to merge this if you'd added tests covering this scenario

I had planned to yes, but you responded so quickly! Thanks in advance.

7ochem commented:
@escopecz I have added tests and updated my branch to include the PHPCS and PHPStan fixes so everything goes green now.

Can you review this? And if you are going to merge it, could you also create a new version/tag?

7ochem commented:
Thanks for merging and tagging @escopecz! Could you give https://packagist.org/packages/escopecz/mautic-form-submit an update please? Thanks!

escopecz commented:
Thanks for the reminder. I thought it's automatic. Had to press a button there.
